packages/flutter/test/material/bottom_sheet_test.dart

```diff
@@ -726,6 +726,115 @@ void main() {
 
     expect(retrievedRouteSettings, routeSettings);
   });
+
+  testWidgets('Verify showModalBottomSheet use AnimationController if provided.',
+          (WidgetTester tester) async {
+    const Key tapTarget = Key('tap-target');
+    await tester.pumpWidget(MaterialApp(
+      home: Scaffold(
+        body: Builder(
+            builder: (BuildContext context) {
+              return GestureDetector(
+                onTap: () {
+                  showModalBottomSheet<void>(
+                    context: context,
+                    transitionAnimationController: AnimationController(
+                        vsync: const TestVSync(),
+                        duration: const Duration(seconds: 2),
+                        reverseDuration: const Duration(seconds: 2)),
+                    builder: (BuildContext context) {
+                      return Container(
+                        child: const Text('BottomSheet'),
+                      );
+                    },
+                  );
+                },
+                behavior: HitTestBehavior.opaque,
+                child: Container(
+                  height: 100.0,
+                  width: 100.0,
+                  key: tapTarget,
+                ),
+              );
+            }
+        ),
+      ),
+    ));
+
+    expect(find.text('BottomSheet'), findsNothing);
+
+    await tester.tap(find.byKey(tapTarget)); //Opening Animation will starts after tapping
+    await tester.pump();
+
+    expect(find.text('BottomSheet'), findsOneWidget);
+    await tester.pumpAndSettle(const Duration(seconds: 2));
+    expect(find.text('BottomSheet'), findsOneWidget);
+
+    // Tapping above the bottom sheet to dismiss it.
+    await tester.tapAt(const Offset(20.0, 20.0)); //Closing Animation will starts after taping
+    await tester.pump();
+
+    expect(find.text('BottomSheet'), findsOneWidget);
+    await tester.pumpAndSettle(const Duration(seconds: 2));
+    expect(find.text('BottomSheet'), findsNothing);
+  });
+
+  testWidgets('Verify persistence BottomSheet use AnimationController if provided.',
+          (WidgetTester tester) async {
+    const Key tapTarget = Key('tap-target');
+    const Key tapTargetToClose = Key('tap-target-to-close');
+    await tester.pumpWidget(MaterialApp(
+      home: Scaffold(
+        body: Builder(
+            builder: (BuildContext context) {
+              return GestureDetector(
+                onTap: () {
```

Comment:
Some of the indentation here and in other spots of the test file are off. Please check the style guide if you are unsure of how to format: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo